File: kotlin/src/001_Two_Sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        for (i in nums.indices) {
            val findValue = target - nums[i]
            for (j in i+1 until nums.size) {
                if (nums[j] == findValue) {
                    return intArrayOf(i, j)
                }
            }
        }
        return intArrayOf()
    }
}

fun main() {
    val solution = Solution()
    val nums = intArrayOf(2, 7, 11, 15)
    val target = 9

    val result = solution.twoSum(nums, target)
    println("Indices: ${result[0]} ${result[1]}")
}

File: python/001_Two_Sum.py
"""
Given an array of integers, return indices of the two numbers such that they add up to a specific target
For example:
Given nums = [2, 7, 11, 15], target = 9,

Because nums[0] + nums[1] = 2 + 7 = 9,
return [0, 1].
"""
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            find_data = target - nums[i]
            idx = i+1

            if find_data in nums[idx:]:
                return [i, nums[idx:].index(find_data)+idx]


if __name__ == '__main__':
	print(Solution().twoSum([2, 7, 11, 15], 9))
